tests/primer/primer_tool.py

```diff
@@ -112,11 +130,79 @@ def _handle_run_command(self) -> None:
         ) as f:
             json.dump(packages, f)
 
+    def _handle_compare_command(self) -> None:
+        with open(self.config.base_file, encoding="utf-8") as f:
+            main_dict: PackageMessages = json.load(f)
+        with open(self.config.new_file, encoding="utf-8") as f:
+            new_dict: PackageMessages = json.load(f)
+
+        final_main_dict: PackageMessages = {}
+        for package, messages in main_dict.items():
+            final_main_dict[package] = []
+            for message in messages:
+                try:
+                    new_dict[package].remove(message)
+                except ValueError:
+                    final_main_dict[package].append(message)
+
+        self._create_comment(final_main_dict, new_dict)
+
+    def _create_comment(
+        self, all_missing_messages: PackageMessages, all_new_messages: PackageMessages
+    ) -> None:
+        comment = ""
+        for package, missing_messages in all_missing_messages.items():
+            new_messages = all_new_messages[package]
+            package_data = self.packages[package]
+
+            if not missing_messages and not new_messages:
+                continue
+
+            comment += f"\n\n**Effect on [{package}]({self.packages[package].url}):**\n"
+
+            if missing_messages:
+                comment += "The following messages are no longer emitted:\n"
+            count = 1
+            for message in missing_messages:
+                comment += (
+                    f"{count}) {message['symbol']} on line {message['line']} here:\n"
+                )
+                filepath = str(message["path"]).replace(
+                    str(package_data.clone_directory), ""
+                )
+                comment += f"{package_data.url}/blob/{package_data.branch}/{filepath}#L{message['line']}\n"
+                count += 1
+            comment += "\n"
+
+            count = 1
+            if new_messages:
+                comment += "The following messages are now emitted:\n"
+            for message in new_messages:
+                comment += (
+                    f"{count}) {message['symbol']} on line {message['line']} here:\n"
+                )
+                filepath = str(message["path"]).replace(
+                    str(package_data.clone_directory), ""
+                )
+                comment += f"{package_data.url}/blob/{package_data.branch}/{filepath}#L{message['line']}\n"
+                count += 1
+
+        if comment == "":
+            comment = "🤖 According to the primer, this change has **no effect** on the checked open source code. 🤖🎉"
+        else:
+            comment = (
+                "🤖 **Effect of this PR on checked open source code:** 🤖\n\n" + comment
+            )
+
+        with open("comment.txt", "w", encoding="utf-8") as f:
```

Comment:
Can we put this under `.pylint_primer_tests`?

Reply:
Yes, although some other stuff will need to move as well.

Btw, I was thinking of moving `. pylint_primer_tests` into `./tests`. It feels out of place in root. What do you think?

Reply:
See https://github.com/PyCQA/pylint/pull/6728